mypy/build.py

```diff
@@ -820,7 +820,7 @@ def remove_cwd_prefix_from_path(p: str) -> str:
 find_module_isdir_cache = {}  # type: Dict[Tuple[str, str], bool]
 
 # Cache packages for each Python executable
-package_dirs_cache = {}  # type: Dict[str, List[str]]
+package_dirs_cache = {}  # type: Dict[Optional[str], List[str]]
```

Comment:
This key type feels odd. Do you really expect/allow `None` as a cache key? What does it mean?